domain-ee/ee-ep-merge-app/src/python_src/api.py

```diff
@@ -67,30 +79,65 @@ async def on_shut_down():
 )
 
 
-@app.get("/health")
+@app.get("/health",
+         response_model=HealthResponse,
+         response_model_exclude_none=True)
 def get_health_status():
-    return {"status": "ok"}
+    if is_healthy():
+        return {"status": "healthy"}
+    else:
+        errors = []
+        if not HOPPY.is_ready():
+            errors.append("Cannot connect to RabbitMQ.")
+        if not JOB_STORE.is_ready():
+            errors.append("Cannot connect to database.")
+        return JSONResponse(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                            content={
+                                "status": "unhealthy",
+                                "errors": errors
+                            })
+
+
+def is_healthy():
+    return HOPPY.is_ready() and JOB_STORE.is_ready()
```

Comment:
I see `get_health_status` and `merge_claims` both using this boolean result by subsequently asking again for `HOPPY.is_ready()` and `JOB_STORE.is_ready()`.

Consider folding that logic into here and having this function (named `health_check_errors` maybe) return a list (possibly empty) of constant plain-English strings.

The strings in `get_health_status` seem to be identical to the constants too.